### ISSUE TITLE
Patroni kubernetes tcp keepalives

### ISSUE THREAD

**Issue by rkojedzinszky** [zalando/patroni#1384]
I was playing with my patroni installation on k8s, and during a test, a patroni nodes could not reach k8s apiserver for a while. When it came back, it remained in an undesired state:
```
2020-02-03 06:36:45,997 INFO: following a different leader because i am not the healthiest node
```

This is because, this node still thinks that it is communicating with the apiserver:
```
# nsenter -n -t 6692 ss -n -o
Netid        State        Recv-Q        Send-Q                         Local Address:Port                           Peer Address:Port                                              
udp          ESTAB        0             0                                      [::1]:44229                                 [::1]:44229                                             
tcp          ESTAB        0             0                                  127.0.0.1:5432                              127.0.0.1:37422         timer:(keepalive,71582min,0)        
tcp          ESTAB        0             0                               10.112.0.156:56268                             10.96.0.1:443                                               
tcp          ESTAB        0             0                                  127.0.0.1:38436                             127.0.0.1:5432          timer:(keepalive,720ms,0)           
tcp          ESTAB        0             0                                  127.0.0.1:37422                             127.0.0.1:5432          timer:(keepalive,93min,0)           
tcp          ESTAB        0             0                               10.112.0.156:56266                             10.96.0.1:443                                               
tcp          ESTAB        0             0                                  127.0.0.1:5432                              127.0.0.1:38436         timer:(keepalive,5.850ms,0)         
tcp          ESTAB        0             0                      [::ffff:10.112.0.156]:9187                  [::ffff:10.112.12.33]:42870         timer:(keepalive,750ms,0)           
```

Here you see that the apiserver connection has no tcp keepalives defined, so perhaps patroni will never notice that the apiserver is back again, nor will try the reconnect again.

I think, it would be really necessary here to have tcp keepalives set for the apiserver connections.

**Comment by CyberDem0n** [comment]
It is already fixed in the 1.6.4:https://github.com/zalando/patroni/pull/1366

**Comment by rkojedzinszky** [comment]
Thanks, I've missed that update. Closing.
